### ISSUE TITLE
incompatible at least with react 16.7.0-alpha.0

### ISSUE THREAD

**Issue by a-x-** [maicki/why-did-you-update#47]
```
TypeError: Class constructor DocheckStatPage cannot be invoked without 'new'
```

```
  var WDYUClassComponent = function (_ctor) {
    _inherits(WDYUClassComponent, _ctor);

    function WDYUClassComponent() {
      _classCallCheck(this, WDYUClassComponent);

      return _possibleConstructorReturn(this, _ctor.apply(this, arguments)); // <<< HERE
    }
```

**Comment by a-x-** [comment]
"@babel/core": "7.0.0-beta.39",

**Comment by vzaidman** [comment]
i think ill get to it when this version will move to beta.
everybody are welcome to contribute before that.

what do you mean by
> "@babel/core": "7.0.0-beta.39"
?

**Comment by a-x-** [comment]
I thanked that Babel version might be helpful too

**Comment by a-x-** [comment]
But latest stable Babel releases behave like so

**Comment by vzaidman** [comment]
ok. i strongly suggest to update babel to the latest 7 version anyway :)

**Comment by knoid** [comment]
I'm getting the same error with React 16.3.1 and babel-core 6.26.3. What else could this be related to?

**Comment by danburzo** [comment]
Sounds like it's because you're not transpiling your class components with Babel for the development environment (fair) and `nwb` is not transpiling class inheritance correctly `class WDYUClassComponent extends YourComponent { }`.

**Comment by danburzo** [comment]
So the idea here (and in #48) is that Babel-transpiled class inheritance cannot work when the base class is not transpiled itself. 

A quick solution that I can suggest is to also publish the `src/` folder on NPM and let people import:

```js
import WDYU from 'why-did-you-update/src';
```

**Comment by vzaidman** [comment]
try this:
https://github.com/welldone-software/why-did-you-render

**Comment by knoid** [comment]
already did, same problem (but only on the enabled components)

**Comment by vzaidman** [comment]
fixed this on my more advanced project on version 2.2.3

"@welldone-software/why-did-you-render": "2.2.3",
check the sandbox:
https://codesandbox.io/s/44qrzy7v0?expanddevtools=1

i would welcome a PR on this project that fixes the same issue.

you can look here to understand how:
https://github.com/welldone-software/why-did-you-render/blob/master/src/whyDidYouRender.js

**Comment by vzaidman** [comment]
more about this: https://overreacted.io/how-does-react-tell-a-class-from-a-function/

**Comment by matthew-dean** [comment]
I discovered that if I use:
```js
const reactRoot = ReactDOM.unstable_createRoot(document.getElementById('root'))

reactRoot.render((
  <ReduxProvider store={configureStore(initialState)}>
    <Authentication>
      <App />
    </Authentication>
  </ReduxProvider>))
```
I get logging as if my App is rendering, and it doesn't trigger the `cannot be invoked without 'new'` error but it never mounts! (DOM is empty.)

**Comment by vzaidman** [comment]
I found out the problem is caused by transpiled class trying to extend native class.
It happens when you build your app with the target of latest browsers.
Targeting latest browsers cause babel-env to not transpile classes.
The library always transpiles classes.
Transpiled classes are failing to extend untranspiled classes.

Fixed this in the `why-did-you-render` library:
https://github.com/welldone-software/why-did-you-render/issues/5
